### PR TITLE
ci(release): inherit secrets so the npm environment token reaches the publisher

### PR DESCRIPTION
**Pull request by javiertoledo**
The 0.3.1 run got further than the last one — `package-release` passed, so #59's fix works — and then split:

| | Result |
|---|---|
| GHCR | ✅ six images promoted at `0.3.1` and `sha-73f88778a5a3` |
| npm | ❌ nothing — `@theagilemonkeys/facility` is absent from the registry |
| tag | ❌ none; `record-release` needs both publishers |

`publish-npm / publish` failed at **Bootstrap the package once with a granular token** with `NPM_BOOTSTRAP_TOKEN is required for the first package publication`, and `NODE_AUTH_TOKEN` printed blank rather than `***` — an empty value, not a masked one.

## Why

The token is fine. `NPM_BOOTSTRAP_TOKEN` exists in the `npm` environment (created 2026-07-31), the environment allows only `main`, and the publish job correctly declares `environment: npm`.

The gap is the reusable-workflow boundary: `ci.yml` calls `release.yml` with no `secrets:` key. A called workflow does not receive environment-scoped secrets by declaring the environment alone — without `secrets: inherit` on the caller, **every environment-scoped secret resolves to an empty string** rather than erroring. GitHub's docs omit this; it is tracked in [actions/runner#1490](https://github.com/actions/runner/issues/1490) and [github/docs#44458](https://github.com/github/docs/issues/44458).

So the job authenticated with nothing and failed the way it would if the secret had never been created.

## Why `inherit` is not a widening here

- The environment still gates the token: it materializes only in the job naming `environment: npm`, which keeps the branch policy and approval as the real control.
- `gh api repos/theam/facility/actions/secrets` and the org-secrets endpoint both return empty — there is nothing else for `inherit` to forward.
- Explicitly mapping `NPM_BOOTSTRAP_TOKEN:` from the caller cannot work: the calling job does not declare the environment, so it would forward an empty string too.

`publish-images` needs no change — it uses `GITHUB_TOKEN`, which called workflows always receive.

## Verification

`node --test scripts/*.test.mjs` → 88/88. The new test asserts `secrets: inherit` on the caller **and** `environment: npm` on the callee; removing either makes it fail, checked both ways. Both workflows parse, and `secrets` is accepted on the `publish-npm` job.

I cannot prove this end to end from a PR — `publish-npm` only runs on a push to `main`, so the release run is again the first real exercise.

## Before merging

The six GHCR images already carry `0.3.1` from `73f8877`. `scripts/images.mjs:160` refuses to move an existing tag to a different digest, and `.dockerignore` does not exclude `.github`, so rebuilding after this commit may produce different digests and wedge `promote`. **Delete the six `0.3.1` package versions first**, so this merge can publish a coherent 0.3.1 to both registries.

## Follow-up worth considering, not included here

`docs/releasing.md` tells maintainers to retry the failed jobs in the same run. That cannot work when the failure is in the workflow itself, since a re-run uses the workflow files from the run's own commit — true for both 0.3.1 attempts. The runbook deserves that caveat.

### CODE REVIEW

File: scripts/release.test.mjs [hunk 1]
```diff
@@ -188,6 +188,28 @@ test("both release validations receive the visibility their policy checks", () =
...
+// published six images and nothing to npm. Both halves are asserted here
+// because either one alone silently reproduces that split.
+test("the npm publisher inherits the secrets its environment gates", () => {
+  const publishNpm = ciWorkflow.split("\n  publish-npm:")[1]?.split("\n\n  ")[0];
+  assert.ok(publishNpm, "CI workflow must contain a publish-npm job");
+  assert.match(publishNpm, /uses: \.\/\.github\/workflows\/release\.yml/);
+  assert.match(
+    publishNpm,
+    /\n    secrets: inherit\b/,
+    "publish-npm must pass secrets to the reusable workflow, or NPM_BOOTSTRAP_TOKEN arrives empty",
```
[the remark this answers is not in the document]
Comment (javiertoledo) [reply scripts/release.test.mjs:204]: Split this one: you found a real gap, and the specific test you asked for cannot be built honestly. Taking them separately.

**The gap — fixed in 419af62.** You are right that the denial path was untested, and it was the exact path the failed release executed: `publishTarball` refuses to run when `NODE_AUTH_TOKEN` is absent, and nothing asserted that refusal. There is now an integration test covering both shapes of "no token" — empty string and unset — against the existing fake registry, asserting the refusal lands *before* any write reaches it. Empty is the important one: a secret the caller never passed arrives as `""` rather than an error, so it is indistinguishable from a configured-but-blank token to every line of code downstream. I verified the test by deleting the guard in `release.mjs`; it fails.

**What I am not adding: a test that the token materializes across the reusable-workflow boundary.** Both available approaches would produce a test that passes while production fails.

- A local runner such as `act` does not model environment-scoped secrets or environment protection the way GitHub does. Concretely, it would not have reproduced this bug — a green run would have told us the wiring was fine while the release failed exactly as it did.
- A hand-rolled fake would encode *our* model of GitHub's secret-scoping rules and then assert against it. But a wrong model of those rules is the whole defect: the belief that `environment: npm` on the called job is sufficient. A fake built on that belief passes; one built on the corrected belief only restates the fix. Either way it verifies the author's understanding, not GitHub's behaviour.

What actually constrains this is that the publisher fails closed, and that is now demonstrated rather than assumed — twice in production, and by the test above. Combined with the unit test asserting `secrets: inherit` on the caller and `environment: npm` on the callee (removing either fails it), the deletable surface is covered. What remains uncoverable is GitHub's own semantics, and I would rather say so than add a green test that implies otherwise.

On the `AGENTS.md` reference: agreed it governs here, which is why the denial-path half was a must-fix rather than a preference.
